http-common/src/connector.rs

```diff
@@ -768,8 +753,59 @@ fn get_num_systemd_sockets() -> Result<Option<std::os::unix::io::RawFd>, String>
         }
     }
 
-    // Return the number of systemd sockets.
-    Ok(Some(listen_fds))
+    // If there is only one socket, we know this is the identity service which uses only one socket, so we have a match:
+    if listen_fds == 1 {
+        return Ok(Some(SD_LISTEN_FDS_START));
+    }
+
+    // If there is more than 1 socket and we don't have a socket name to match, this is edged telling us that there is no systemd socket we can match.
+    let socket_name = match socket_name {
+        Some(socket_name) => socket_name,
+        None => return Ok(None),
+    };
+
+    // If there is more than one socket, this is edged. We can attempt to match the socket name to systemd.
+    // This happens when a unix Uri is provided in the config.toml. Systemd sockets get created nonetheless, so we still prefer to use them.
+    // If a socket name is provided but we don't see the env variable LISTEN_FDNAMES, it means we are probably on an older OS, and we can't match either.
+    let listen_fdnames = match get_env("LISTEN_FDNAMES")? {
+        Some(listen_fdnames) => listen_fdnames,
+        None => return Ok(None),
+    };
+    let listen_fdnames: Vec<&str> = listen_fdnames.split(':').collect();
+
+    #[allow(
+        clippy::cast_possible_truncation,
+        clippy::cast_sign_loss,
+        clippy::cast_possible_wrap
+    )]
+    let len: std::os::unix::io::RawFd = std::convert::TryInto::try_into(listen_fdnames.len())
+        .expect("usize -> std::os::unix::io::RawFd");
+    if listen_fds != len {
+        return Err(format!(
+            "Mismatch, there are {} fds, and {} names",
+            listen_fds,
+            listen_fdnames.len()
+        ));
+    }
+
+    if let Some(index) = listen_fdnames
+        .iter()
+        .position(|fdname| (*fdname).eq(&socket_name))
+    {
+        #[allow(
+            clippy::cast_possible_truncation,
+            clippy::cast_sign_loss,
+            clippy::cast_possible_wrap
+        )]
+        let index: std::os::unix::io::RawFd =
+            std::convert::TryInto::try_into(index).expect("usize -> std::os::unix::io::RawFd");
```

Comment:
`map_err(|_| "invalid number of sockets".to_string())?` instead of `expect`.

Reply:
Ok, I was just copying code I found elsewhere:
let digest_len =
                        std::convert::TryInto::try_into(digest.len()).expect("usize -> c_ulong");